components/d2l-activity-editor/d2l-activity-content-editor/scorm/d2l-activity-content-scorm-activity-detail.js

```
@@ -0,0 +1,157 @@
+import { LocalizeActivityEditorMixin } from '../../mixins/d2l-activity-editor-lang-mixin.js';
+import { MobxLitElement } from '@adobe/lit-mobx';
+import { RtlMixin } from '@brightspace-ui/core/mixins/rtl-mixin.js';
+import { SkeletonMixin } from '@brightspace-ui/core/components/skeleton/skeleton-mixin.js';
+
+class ContentScormActivityDetail extends SkeletonMixin(ErrorHandlingMixin(LocalizeActivityEditorMixin(EntityMixinLit(RtlMixin(ActivityEditorMixin(MobxLitElement)))))) {
+
+	static get properties() {
+		return {
+			activityUsageHref: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [
+			super.styles,
+			labelStyles,
+			activityContentEditorStyles,
+			css`
+				.d2l-activity-label-container {
+					margin-bottom: 6px;
+				}
+				d2l-activity-content-external-activity {
+					margin-top: 36px;
+				}
+			`
+		];
+	}
+
+	constructor() {
+		super(contentScormActivityStore);
+		this.skeleton = true;
+		this._setEntityType(ContentScormActivityEntity);
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		const event = new CustomEvent('d2l-request-provider', {
+			detail: { key: 'd2l-provider-create-selectbox-grade-item-enabled' },
+			bubbles: true,
+			composed: true,
+			cancelable: true
+		});
+		this.dispatchEvent(event);
+
+		this._createSelectboxGradeItemEnabled = event.detail.provider;
+
+		this.saveTitle = this.saveTitle.bind(this);
+		this.saveLinkOptions = this.saveLinkOptions.bind(this);
+		this.saveGradeOptions = this.saveGradeOptions.bind(this);
+	}
+
+	render() {
+		const scormActivityEntity = contentScormActivityStore.getContentScormActivity(this.href);
+
+		let title = '';
+		let subTitle = '';
+
+		if (scormActivityEntity) {
+			this.skeleton = false;
+			title = `${this.localize('content.scormActivity')}: ${scormActivityEntity.title}`;
+			subTitle = `${this.localize('content.lastEdited')} ${new Date(Date.parse(scormActivityEntity.lastEdited)).toUTCString()}`;
```

What happens if lastEdited/lastModified is null: https://github.com/BrightspaceHypermediaComponents/siren-sdk/blob/dad38e58c5f273b75221e4db5cff000f7f8b618e/src/activities/content/ContentScormActivityEntity.js#L35?

---

That would be bad, there should probably be some basic error handling here eh? Suggestions as to what to put in that case?

---

I'm not sure, I suppose it would depend on what the design looks like. One sec.

---

I feel we should have it set up such that we can rely on the Siren Entity always having this value. i.e. if there should always be a value for this, then we should ensure that this gets set properly in the Siren-SDK, and not worry about this here. It would only be a concern if there is actually a non-default value possible, i.e. if Scorm Packages can possibly not have a lastEdited date. Hopefully this makes sense?

Thoughts @ianlaidlaw ?

---

I agree that it should always be there from the HM response which is nice to rely on, I just can't think of what should be displayed in the unlikely case that it isn't coming back properly. A default date doesn't make sense as it would probably seem really weird to have a unrelated date. Maybe just the current date/time if the property doesn't exist? I think if that's the go-ahead it should be done here (instead of the siren-sdk, as to not modify the HM response outside the usage context, but that's just a personal preference. Happy to change whatever people think is the best way.

---

I feel any kind of default value or current time would be misleading. I feel if it always exists and by default is the time the package was first added or perhaps it's a property in the package itself, then it doesn't matter. If we are considering the scenario where for some reason we indeed have a null, I feel the best solution would be not to show anything, i.e. don't show `Last Modified...` at all.

---

I like that idea, let's do that.

---

Addressed in 2edb4274064e294ba0ed46ced61f4be65376d329